### DISCUSSION TITLE
hardhat arbitrum inconsistent test results

### DISCUSSION THREAD

**Discussion by mattador1** [NomicFoundation/hardhat#5010] [Troubleshooting]
Hi, do not know what this could be, basically I have mocha tests and testing it with forked arbitrum with alchemy as provider.
Problem is that tests do not provide me consistent results. If I have many sections "describe" and running one after another I get sometimes faulty tests. What helps is to do npx hardhat clean, isolate faulty "describe" section and run only that describe section. Then it passes. Really strange I must say that before each describe section I do hardhat reset. So always are same conditions. But if there are multiple describe sections one after another and even before each hardhat reset is done, inconsistent results occur. I did not find those problem when forking mainnet. Do you guys know what could be. Thank you in advance.

**Comment by kanej** [comment]
Hello @mattador1, can we confirm, are you seeing inconsistent test results since you updated to the latest version of Hardhat (so anything after `v2.21.0`).

**Reply by mattador1** [reply]
Hi, we are in process of migrating the project to new version of hardhat, but also the whole toolbox. On fork mainnet we did not notice those problems, but when we switched on fork arbitrum, that is when these errors started. Currently we are stuck on 2.16.0. hardhat I will update on the results when we migrate. Thank you for your support.

**Reply by kanej** [reply]
Hardhat `v2.21.0` introduce a major reworking of our internal Hardhat node (a complete rewrite in rust). We have been releasing bug fixes as they get reported; any issues or bugs you encounter we would be keen to hear about. Thanks.
